interface/ceed-qfunction.c

```diff
@@ -743,7 +743,9 @@ int CeedQFunctionAddInput(CeedQFunction qf, const char *field_name, CeedInt size
     CeedCheck(strcmp(field_name, qf->input_fields[i]->field_name), qf->ceed, CEED_ERROR_MINOR, "QFunction field names must be unique");
   }
   for (CeedInt i = 0; i < qf->num_output_fields; i++) {
+    // LCOV_EXCL_START
     CeedCheck(strcmp(field_name, qf->output_fields[i]->field_name), qf->ceed, CEED_ERROR_MINOR, "QFunction field names must be unique");
+    // LCOV_EXCL_STOP
   }
```

Comment:
Is this never called because we always set inputs before outputs? (thus `num_output_fields` is always zero and the loop never iterates)

If that's true, would it be better to flip the input/output calling order of a test rather than exclude this section of code?

Reply:
Yeah.

6 of one, half dozen of the other. We could move the loop, but why always do more work?

Reply:
I don't follow what you mean by moving the loop. I'm not suggesting changing the function, just a test. Like for t523, change:
https://github.com/CEED/libCEED/blob/ed5267d9dde71d634ebc200fed11e1feb22e46ab/tests/t523-operator.c#L87-L90

to

```c
  CeedQFunctionCreateInterior(ceed, 1, mass, mass_loc, &qf_mass_tet);
  CeedQFunctionAddInput(qf_mass_tet, "rho", 1, CEED_EVAL_NONE);
  CeedQFunctionAddOutput(qf_mass_tet, "v", 1, CEED_EVAL_INTERP);
  CeedQFunctionAddInput(qf_mass_tet, "u", 1, CEED_EVAL_INTERP);
```